libraries/helpers_windows.rb

```diff
@@ -56,31 +58,35 @@ def build_rule(new_resource)
         type = to_type(new_resource)
         parameters = {}
 
-        parameters['description'] = "\"#{new_resource.description}\""
-        parameters['dir'] = new_resource.direction
-
-        new_resource.program && parameters['program'] = new_resource.program
-        parameters['service'] = new_resource.service ? new_resource.service : 'any'
-        parameters['protocol'] = new_resource.protocol
-
-        if new_resource.direction.to_sym == :out
-          parameters['localip'] = new_resource.source ? fixup_cidr(new_resource.source) : 'any'
-          parameters['localport'] = new_resource.source_port ? port_to_s(new_resource.source_port) : 'any'
-          parameters['interfacetype'] = new_resource.interface ? new_resource.interface : 'any'
-          parameters['remoteip'] = new_resource.destination ? fixup_cidr(new_resource.destination) : 'any'
-          parameters['remoteport'] = new_resource.dest_port ? port_to_s(new_resource.dest_port) : 'any'
+        if type == :log
+          "set currentprofile logging #{new_resource.logging} enable"
```

Comment:
This is exactly what should be done when someone coded logging attribute in a provider call. This is the only way to change windows firewall logging. 

And logging is not per rule in Windows Firewall. It is system wide. You either define logging or not, globally.